citesphere/src/main/java/edu/asu/diging/citesphere/web/user/jobs/ImportCollectionsController.java

```diff
@@ -0,0 +1,149 @@
+package edu.asu.diging.citesphere.web.user.jobs;
+
+import java.io.IOException;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.social.zotero.exception.ZoteroConnectionException;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import edu.asu.diging.citesphere.api.v1.model.impl.Collections;
+import edu.asu.diging.citesphere.api.v1.user.JsonUtil;
+import edu.asu.diging.citesphere.core.exceptions.GroupDoesNotExistException;
+import edu.asu.diging.citesphere.core.exceptions.ZoteroItemCreationFailedException;
+import edu.asu.diging.citesphere.core.model.jobs.IUploadJob;
+import edu.asu.diging.citesphere.core.service.ICitationCollectionManager;
+import edu.asu.diging.citesphere.core.service.ICitationManager;
+import edu.asu.diging.citesphere.core.service.IGroupManager;
+import edu.asu.diging.citesphere.core.service.jobs.IUploadCollectionJobManager;
+import edu.asu.diging.citesphere.model.bib.ICitationCollection;
+import edu.asu.diging.citesphere.model.bib.ICitationGroup;
+import edu.asu.diging.citesphere.user.IUser;
+import edu.asu.diging.citesphere.user.impl.User;
+
+@Controller
+public class ImportCollectionsController {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    
+    @Autowired
+    private IUploadCollectionJobManager jobManager;
+    
+    @Autowired
+    private ICitationManager citationManager;
+    
+    @Autowired
+    private ICitationCollectionManager collectionManager;
+
+    @Autowired
+    private IGroupManager groupManager;
+
+    @Autowired
+    private JsonUtil jsonUtil;
+    
+    @RequestMapping(value = "/auth/import/collection", method = RequestMethod.GET)
+    public String show(Model model, Authentication authentication) {
+        model.addAttribute("groups", citationManager.getGroups((IUser)authentication.getPrincipal()));
+        return "auth/import/collection";
+    }
+    
+    @RequestMapping(value = "/auth/import/collection/getgroupcollections", method = RequestMethod.GET)
```

Comment:
this method is just for getting all collections of a group, right? we probably can't use the api collections controller because of the way we authenticate (if we can, that should be used), but there should be a separate controller (maybe in an api package within `web/user`) that has a method to get the collections.